src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -336,18 +336,39 @@ internal virtual LexicalSortKey GetLexicalSortKey()
 
 #nullable enable
 
-        public virtual Location? TryGetFirstLocation()
```

Comment:
>public virtual Location? TryGetFirstLocation() [](http://example.com/codeflow?start=8&length=46)

In order to reduce the diff, I think we should keep the method, but it would be fine to make it non-virtual and implement it in terms of the new APIs.

Reply:
Removal of this API might also cause an unnecessary distraction/complication for the purpose of integrating between branches.

Reply:
📝 Note that this change is implemented entirely in its own commit.

Reply:
> Note that this change is implemented entirely in its own commit.

This makes no difference. Please restore all call sites ones this API is restored.

Reply:
> I'll break it apart by rebasing near the end. For now the ongoing work from Cyrus is making it difficult to stay on top of it.

I am not sure how to interpret this. Could you please elaborate?

Reply:
➡️ Prepared the `FirstOrDefault` change as a follow-up #67890

Reply:
@sharwell In order to avoid any confusion, I am objecting to the removal of this API and touching its call sites in general (not just in context of this PR).

Reply:
@AlekseyTs That's fine, I have no plans to bring it back in this PR and y'all can decide either way on the other one.